src/fastoad/openmdao/problem.py

```diff
@@ -77,28 +85,56 @@ def setup(self, *args, **kwargs):
         """
         Set up the problem before run.
         """
-        with problem_without_mpi(self) as problem_copy:
-            try:
-                super(FASTOADProblem, problem_copy).setup(*args, **kwargs)
-            except RuntimeError:
-                vars_metadata = self._get_undetermined_dynamic_vars_metadata(problem_copy)
-                if vars_metadata:
-                    # If vars_metadata is empty, it means the RuntimeError was not because
-                    # of dynamic shapes, and the incoming self.setup() will raise it.
-                    ivc = om.IndepVarComp()
-                    for name, meta in vars_metadata.items():
-                        # We use a (2,)-shaped array as value here. This way, it will be easier
-                        # to identify dynamic-shaped data in an input file generated from current
-                        # problem.
-                        ivc.add_output(name, [np.nan, np.nan], units=meta["units"])
-                    self.model.add_subsystem(SHAPER_SYSTEM_NAME, ivc, promotes=["*"])
+        self.self_analysis()
+
+        self._analysis.fills_dynamically_shaped_inputs(self)
```

Comment:
Isn't this line already ran when you do a `ProblemAnalysis` ?

Reply:
I had hesitated making a property for that, that would control the running of the analysis at the first usage.

Your question made me decide it was the way to go. Modification done in 4eeb488a828b3d679ec208c792b7f1034be563ae.